atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -305,6 +308,7 @@ protected CallbackAwareTransaction createTransaction(
             LongSupplier startTimestampSupplier,
             LockToken immutableTsLock,
             PreCommitCondition condition) {
+        Optional<LockToken> immutableTimestampLock = Optional.of(immutableTsLock);
```

Comment:
Extract a small ImmutableTimestampLock holder?

Reply:
This will be coming soon (planned in the pre-commit "set of things" to merge)!